system/jlib/jsecrets.cpp

```diff
@@ -1308,6 +1250,117 @@ jlib_decl bool containsEmbeddedKey(const char *certificate)
     return false;
 }
 
+
+//---------------------------------------------------------------------------------------------------------------------
+
+class CIssuerConfig final : public CInterfaceOf<ISyncedPropertyTree>
+{
+public:
+    CIssuerConfig(const char *_issuer, const char * _trustedPeers, bool _acceptSelfSigned)
+    : issuer(_issuer), trustedPeers(_trustedPeers), acceptSelfSigned(_acceptSelfSigned)
+    {
+        secret.setown(resolveSecret("certificates", issuer, nullptr, nullptr));
+        createConfigFromSecret();
+    }
+
+    virtual const IPropertyTree * getTree() const override;
+
+    virtual bool getProp(MemoryBuffer & result, const char * key) const override
+    {
+        CriticalBlock block(secretCs);
+        checkStale();
+        return getSecretKeyValue(result, config, key);
+    }
+    virtual bool getProp(StringBuffer & result, const char * key) const override
+    {
+        CriticalBlock block(secretCs);
+        checkStale();
+        return getSecretKeyValue(result, config, key);
+    }
+    virtual bool isStale() const override
+    {
+        return secret->isStale();
+    }
+    virtual unsigned getVersion() const override
+    {
+        //If information that is combined with the secret (e.g. trusted peers) can also change dynamically this would
+        //need to be a separate hash calculated from the condig tree
+        return secretHash;
+    }
+
+protected:
+    void checkStale() const;
+    void createConfigFromSecret() const;
+
+protected:
+    StringAttr issuer;
+    StringAttr trustedPeers;
+    bool addCACert = true;
+    bool acceptSelfSigned = false;
+    Owned<ISyncedPropertyTree> secret;
+    mutable CriticalSection secretCs;
+    mutable Linked<IPropertyTree> config;
+    mutable unsigned secretHash = 0;
+};
+
+
+const IPropertyTree * CIssuerConfig::getTree() const
+{
+    CriticalBlock block(secretCs);
+    checkStale();
+    return LINK(config);
+}
+
+void CIssuerConfig::checkStale() const
+{
+    CriticalBlock block(secretCs);
+    if (secretHash != secret->getVersion())
+        createConfigFromSecret();
+}
+
+void CIssuerConfig::createConfigFromSecret() const
+{
+    config.clear();
+
+    //Update before getting the tree to avoid potential race condition updating the tree at the same time.
+    //Could alternatively return the version number from the getTree() call.
+    secretHash = secret->getVersion();
+    Owned<const IPropertyTree> secretInfo = secret->getTree();
+
+    if (secretInfo)
+    {
+        StringBuffer value;
+        config.setown(createPTree(issuer));
+
+        //GH-Tony should this condition be removed?
+        if (strieq(issuer, "remote")||strieq(issuer, "local"))
```

Comment:
It looks like you've mixed some "ClientConfig" lines in with "ServerConfig" lines.
When inside ClientConfig these lines imply that we don't currently setup issuer client certificates for issuers other than remote and local.
Using this logic on the ServerConfig won't work, because server side we always want the certs included.